presto-jdbc/src/main/java/com/facebook/presto/jdbc/ParameterInfo.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.jdbc;
+
+import com.facebook.presto.spi.type.TypeSignature;
+
+import static java.util.Objects.requireNonNull;
+
+public class ParameterInfo
+        extends AbstractInfo
+{
+    private final int position;
+    private final int parameterType;
+    private final TypeSignature parameterTypeSignature;
+    private final Nullable nullable;
+    private final boolean signed;
+    private final int precision;
+    private final int scale;
+
+    public ParameterInfo(
+            int position,
+            int parameterType,
+            TypeSignature parameterTypeSignature,
+            Nullable nullable,
+            boolean signed,
+            int precision,
+            int scale)
+    {
+        this.position = position;
+        this.parameterType = parameterType;
+        this.parameterTypeSignature = requireNonNull(parameterTypeSignature, "parameterTypeName is null");
+        this.nullable = requireNonNull(nullable, "nullable is null");
+        this.signed = signed;
+        this.precision = precision;
+        this.scale = scale;
+    }
+
+    public static void setTypeInfo(ParameterInfo.Builder builder, TypeSignature type)
```

Comment:
It would be nice to avoid duplicating this between `ColumnInfo` and `ParameterInfo`. I'm thinking we could extract a `TypeInfo` class that would be used as a member of `ColumnInfo` and `ParameterInfo`. It would have these fields:
```java
private final int sqlType;
private final List<Integer> parameterSqlTypes;
private final TypeSignature typeSignature;
private final Nullable nullable;
private final boolean currency;
private final boolean signed;
private final int precision;
private final int scale;
private final int displaySize;
```
`parameterSqlTypes` and `displaySize` are only needed for `ColumnInfo`, but it doesn't hurt to have them. `parameterSqlTypes` is the `columnParameterTypes` field from `ColumInfo` and contains the parameter types for arrays or other nested types.

After that, `ColumInfo` would only need these fields:

```java
private final TypeInfo typeInfo;
private final String columnLabel;
private final String columnName;
private final String tableName;
private final String schemaName;
private final String catalogName;
```

What do you think of this approach?

Reply:
I thought about this redundancy as well, so your idea is suitable and I will implement it.